src/OpenTelemetry/Metrics/MetricAggregators/SumMetricAggregator.cs

```diff
@@ -60,33 +56,11 @@ public IDataValue Sum
             {
                 if (this.valueType == typeof(long))
                 {
-                    long sum;
-
-                    if (this.IsMonotonic)
-                    {
-                        sum = this.sumPos + (long)this.dsumPos;
-                    }
-                    else
-                    {
-                        sum = this.sumPos + (long)this.dsumPos + this.sumNeg + (long)this.dsumNeg;
-                    }
-
-                    return new DataValue(sum);
+                    return new DataValue(this.sumLong);
```

Comment:
the `public IDataValue Sum` is not thread-safe. We may avoid that issue if we refactor to avoid SumMetricAggregator implementing both ISumMetric and IAggregator intefaces. The latter represents the thing which does the aggregation, and former is the "aggregation output". Will tackle this in another PR.